test/test_fork.py

```diff
@@ -98,3 +100,42 @@ def test_topology_reset(self):
             self.assertNotEqual(child_id, init_id)
             passed, msg = parent_conn.recv()
             self.assertTrue(passed, msg)
+
+    def test_many_threaded(self):
+        # Fork randomly while doing operations.
+        class ForkThread(threading.Thread):
+            def __init__(self, runner):
+                self.runner = runner
+                super().__init__()
+
+            def run(self) -> None:
+                clients = []
+                for _ in range(10):
+                    clients.append(MongoClient())
+
+                # The sequence of actions should be somewhat reproducible.
+                # If truly random, there is a chance we never actually fork.
+                # The scheduling is somewhat random, so rely upon that.
+                def action(client):
+                    client.admin.command("ping")
+                    return 0
+
+                for i in range(200):
+                    # Pick a random client.
+                    rc = clients[i % len(clients)]
+                    if i % 50 == 0:
+                        # Fork
+                        pid = os.fork()
+                        if pid == 0:  # Child => Can we use it?
+                            code = action(rc)
+                            os._exit(code)
```

Comment:
If the action() fails then `os._exit(code)` will never run. I think we need to call os._exit in a finally block (for all the tests that use fork).